### ISSUE TITLE
No stable GPS fix, no FLARM out:  Pi3B Stratux with T-Beam

### ISSUE THREAD

**Issue by polarjet** [VirusPilot/stratux-pi4#21]
My goal: Stratux (Pi3B) with Flarm OUT (T-Beam via SoftRF)
My devices/Software installed yesterday:
- Pi3B:  
Stratux v1.6r1-eu29-vp (f716aaa962)  - installed with no problems acc.  https://github.com/VirusPilot/stratux-pi4
- T-Beam T22_v1.1 20210222 with GNSS U8, Radio SX127x, Baro BMP280
this T-Beam was preiinstalled with SoftRF 1.2 ESP32 from around 10month ago
Did yesterday SOftRF Compiling/Flashing from Source Code with Arduino acc. https://github.com/VirusPilot/SoftRF

Issues now:
- hard to get a GPS fix (having external GPS Ant and being outside with free sky). Getting sometimes a fix but looses fix quickly, not stable at all --  so no Flarm Tx packets 
-  standard T-Beam WIFI password 123...678  not working anymore. So cannot configure T-Beam.

Please can you help ?

**Comment by VirusPilot** [comment]
- Did you have a good GPS fix before compiling/flashing?
- The U.FL connectors are very sensitive so did you check the GPS antenna connection?
- In my SoftRF fork I have not changed the WiFi so it should work right away. Can you connect to the SoftRF WiFi at all?
- Instead of compiling/flashing you can use [https://github.com/VirusPilot/SoftRF?tab=readme-ov-file#binaries-unstable-beta-versions-available-for-testing](url).

**Comment by polarjet** [comment]
Regarding GPS fix, sorry my fault - the screw of the GPS antenna cable was not tightened fully to the housing nut. Now stable GPS fix.
After adjusting stratux config like attached (disabled ogn transmission via I2C) Flarm is doing now TX :-)
Forgot one stratux issue: I changed IP of stratux to 192.168.1.1 and afterwards pressing " submit WiFi changes", but after restart still old IP  192.168.10.1 is active !?
As of now I cannot connect to T-Beam via WiFi at all. Trying now with flashing T-Beam with your mentioned beta version. Hope WiFi password issue can be solved then.

![Screenshot_2024-02-13-23-19-52-98_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/VirusPilot/stratux-pi4/assets/124938725/9da43c21-9da2-4993-bd6e-21d2d279d419)
![Screenshot_2024-02-13-22-58-02-51_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/VirusPilot/stratux-pi4/assets/124938725/07cfcf58-0d7d-48b5-8dcf-cd8c177dfe18)

**Comment by polarjet** [comment]
Flashing with last beta solved the WiFi issue with t-Beam. Thanks !!
Only open point the stratux IP switch not working.

**Comment by VirusPilot** [comment]
Good to hear. Did you use RaspiOS Bookworm as base system? This is currently not yet supported, I noticed also myself a few days ago that the IP change has no effect so for the time being you should use Bullseye ...

**Comment by VirusPilot** [comment]
One more thing: your OGN noise is quite high, can you send a picture of the hardware setup?

**Comment by VirusPilot** [comment]
Btw, just added a bookworm script based on the standard version. Still WIP ...

**Comment by polarjet** [comment]
> Good to hear. Did you use RaspiOS Bookworm as base system? This is currently not yet supported, I noticed also myself a few days ago that the IP change has no effect so for the time being you should use Bullseye ...

Oh yes, used newest ...Bookworm. Now installing Bullseye Lite 64bit..now via RJ45 cable... Stratux WiFi after using Raspi Imager under Linux Ubuntu not occurring after 10min...twice tried.

**Comment by polarjet** [comment]
> One more thing: your OGN noise is quite high, can you send a picture of the hardware setup?

Yes, I also wanted to ask this..was wondering long time about this. Here my HW setup.
![IMG_20240214_224340](https://github.com/VirusPilot/stratux-pi4/assets/124938725/ec966934-b970-479b-adfd-a0b5ef5a5313)
![Uploading IMG_20240214_224458.jpg…]()

**Comment by polarjet** [comment]
> One more thing: your OGN noise is quite high, can you send a picture of the hardware setup?

Yes, I also wanted to ask this..was wondering long time about this. Here my HW setup:

![IMG_20240214_224340](https://github.com/VirusPilot/stratux-pi4/assets/124938725/caac6ed4-0314-498a-bc52-163701d2e5cf)
![IMG_20240214_224458](https://github.com/VirusPilot/stratux-pi4/assets/124938725/e0a5d8ef-6521-4e6f-8450-92c71b9fbd68)

**Comment by VirusPilot** [comment]
> > Good to hear. Did you use RaspiOS Bookworm as base system? This is currently not yet supported, I noticed also myself a few days ago that the IP change has no effect so for the time being you should use Bullseye ...
> 
> Oh yes, used newest ...Bookworm. Now installing Bullseye Lite 64bit..now via RJ45 cable... Stratux WiFi after using Raspi Imager under Linux Ubuntu not occurring after 10min...twice tried. Bullseye build done. Now able to switch IP address👍🏼GPS fix goes fast. In log I found only these IMU I/O errors:
> 
> ![Screenshot_2024-02-14-23-26-09-50_40deb401b9ffe8e1df2f1cc5ba480b12](https://private-user-images.githubusercontent.com/124938725/304897103-e07c6a2e-a884-4a01-a1e2-df570e749457.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5OTYzMTksIm5iZiI6MTcwNzk5NjAxOSwicGF0aCI6Ii8xMjQ5Mzg3MjUvMzA0ODk3MTAzLWUwN2M2YTJlLWE4ODQtNGEwMS1hMWUyLWRmNTcwZTc0OTQ1Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQxMTIwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjQzYmQwNDNlYTZhZmQ1ODkxNjQxNTY4NGRhOWZkMjdhNjAxMzYxOThiZDkxODlkNTc1NGQwMzlmMzg4NzA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.38bB4L56V5ouOeh_mbBiY7X3QAg1EZlZfvsSAGgLaJY)

You should disable AHRS and BARO Sensor on the Stratux as the Baro values are provided by the T-Beam.

**Comment by VirusPilot** [comment]
Your setup is quite standard but sometimes just the USB power supply makes a difference - reducing noise is an art of its own. Also the fan can cause noise ...

**Comment by VirusPilot** [comment]
btw, no ADSB receiver SDR?

**Comment by polarjet** [comment]
> btw, no ADSB receiver SDR?

Sure i have ADS-B SDR, within this Pic i Had removed it for beeping able to plug in the RJ45 network connector.

**Comment by polarjet** [comment]
> Your setup is quite standard but sometimes just the USB power supply makes a difference - reducing noise is an art of its own. Also the fan can cause noise ...

I did experiments to decrease noise. I found that the USB-extenders (see arrow in pic) are a noise source. I removed the extenders and the noise decreased from 29db on around 14db. My fan active/not active makes no difference in noise.
So I think with 14db noise I can be satisfied. With your hardware, which lowest noise you achieve ?
The log errors are disappeared too after disabling stratux Baro Sensor.
Thank's a lot for your help!! Think we can close this GitHub issue as solved. I feel the software is now much more stable than one year ago.
If you are passing EDTE make a free landing for a coffee :-)
Viele Grüße Martin Zd.

![IMG_20240215_225814](https://github.com/VirusPilot/stratux-pi4/assets/124938725/0e506aac-ac42-4b9b-a1f5-9bb280d17a0f)

![Screenshot_2024-02-15-22-41-02-97_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/VirusPilot/stratux-pi4/assets/124938725/b7a5f954-5369-4632-8383-84cc396ee88c)
